tests/middleware/tests.py

```diff
@@ -62,9 +62,22 @@ def test_append_slash_slashless_unknown(self):
         self.assertEqual(CommonMiddleware().process_response(request, response), response)
 
     @override_settings(APPEND_SLASH=True)
-    def test_append_slash_redirect(self):
+    def test_append_slash_redirect_request(self):
         """
-        APPEND_SLASH should redirect slashless URLs to a valid pattern.
+        Regression test for #26293
+        APPEND_SLASH should redirect slashless URLs in process request
+        to a valid pattern regardless of whether we are prepending www
+        """
+        request = self.rf.get('/slash')
+        r = CommonMiddleware().process_request(request)
+        self.assertEqual(r.status_code, 301)
+        self.assertEqual(r.url, '/slash/')
+
+    @override_settings(APPEND_SLASH=True)
+    def test_append_slash_redirect_response(self):
```

Comment:
As above, since `process_request()` returns a response, the second test isn't needed (see 434d309ef6dbecbfd2b322d3a1da78aa5cb05fa8).